app/controllers/omniauth_controller.rb

```diff
@@ -1,5 +1,6 @@
 class OmniauthController < Devise::OmniauthCallbacksController
   skip_before_action :verify_authenticity_token, only: %i[tra_openid_connect]
+  skip_before_action :authenticate_user!, only: %i[tra_openid_connect failure]
```

Comment:
this is because `Devise::OmniauthCallbacksController` inherits from `ApplicationController`